devito/core/cpu.py

```diff
@@ -104,42 +134,83 @@ def _specialize_iet(cls, graph, **kwargs):
+    def _make_clusters_passes_mapper(cls, **kwargs):
         options = kwargs['options']
-        platform = kwargs['platform']
 
-        blocker = Blocker(options['blockinner'],
-                          options['blocklevels'] or cls.BLOCK_LEVELS)
+        inner = options['blockinner']
+        levels = options['blocklevels'] or cls.BLOCK_LEVELS
+
+        return {
+            'toposort': Toposort().process,
+            'fuse': fuse,
+            'blocking': Blocking(inner, levels).process,
+            # Pre-baked composite passes
+            'topofuse': lambda i: fuse(Toposort().process(i))
+        }
+
+    @classmethod
+    def _make_iet_passes_mapper(cls, **kwargs):
+        options = kwargs['options']
+        platform = kwargs['platform']
 
         ompizer = Ompizer()
 
         return {
-            'denormals': partial(avoid_denormals),
-            'optcomms': partial(optimize_halospots),
-            'wrapping': partial(loop_wrapping),
-            'blocking': partial(blocker.make_blocking),
-            'openmp': partial(ompizer.make_parallel),
+            'denormals': avoid_denormals,
+            'optcomms': optimize_halospots,
+            'wrapping': loop_wrapping,
+            'blocking': partial(relax_incr_dimensions, counter=generator()),
+            'openmp': ompizer.make_parallel,
             'mpi': partial(mpiize, mode=options['mpi']),
             'simd': partial(ompizer.make_simd, simd_reg_size=platform.simd_reg_size),
-            'prodders': partial(hoist_prodders)
+            'prodders': hoist_prodders
         }
 
+    @classmethod
+    def _build(cls, expressions, **kwargs):
+        # Sanity check
+        passes = as_tuple(kwargs['mode'])
+        if any(i not in cls._known_passes for i in passes):
+            raise InvalidOperator("Unknown passes `%s`" % str(passes))
+
+        return super(CustomOperator, cls)._build(expressions, **kwargs)
+
+    @classmethod
+    @timed_pass(name='specializing.Clusters')
+    def _specialize_clusters(cls, clusters, **kwargs):
+        passes = as_tuple(kwargs['mode'])
+
+        # Fetch passes to be called
+        passes_mapper = cls._make_clusters_passes_mapper(**kwargs)
+
+        # Call passes
+        for i in passes:
+            try:
+                clusters = passes_mapper[i](clusters)
+            except KeyError:
+                pass
+
+        return clusters
+
     @classmethod
     @timed_pass(name='specializing.IET')
     def _specialize_iet(cls, graph, **kwargs):
         options = kwargs['options']
         passes = as_tuple(kwargs['mode'])
 
         # Fetch passes to be called
-        passes_mapper = cls._make_passes_mapper(**kwargs)
+        passes_mapper = cls._make_iet_passes_mapper(**kwargs)
 
         # Call passes
         for i in passes:
             try:
                 passes_mapper[i](graph)
             except KeyError:
-                raise InvalidOperator("Unknown passes `%s`" % str(passes))
+                pass
```

Comment:
Why is it passing now? Shouldn't it raise a warning at least?

Reply:
AFAIR, because now this is checked in the constructor (so well before reaching this point)